gtk/gtk-play.c

```diff
@@ -1430,6 +1431,33 @@ media_info_updated_cb (GstPlayer * player, GstPlayerMediaInfo * media_info,
   }
 }
 
+static gboolean
+volume_notify_idle_cb (GtkPlay * play)
+{
+  gdouble new_val, cur_val;
+
+  cur_val = gtk_scale_button_get_value
+    (GTK_SCALE_BUTTON (play->volume_button));
+  new_val = gst_player_get_volume (play->player);
+
+  if (fabs (cur_val - new_val) > 0.001) {
+    g_signal_handlers_block_by_func (play->volume_button,
+        volume_changed_cb, play);
+    gtk_scale_button_set_value (GTK_SCALE_BUTTON (play->volume_button),
+        new_val);
+    g_signal_handlers_unblock_by_func (play->volume_button,
+        volume_changed_cb, play);
+  }
+
+  return FALSE;
+}
+
+static void
+player_volume_changed_cb (GstPlayer * player, GtkPlay * play)
+{
+  g_idle_add ((GSourceFunc) volume_notify_idle_cb, play);
```

Comment:
The loop is the same anyway as the GTK app uses dispatch-to-main-context=TRUE on GstPlayer :)